### DISCUSSION TITLE
Force trailing slash

### DISCUSSION THREAD

**Discussion by galaczi** [adonisjs/core#3600] [Help]
Is there a way to force trailing slash for URLs? E.g. `/post` is not equal to `/post/` although Adonis handles them as the same.

I have this named route:
```
Route.get('/signup/', 'AuthController.signupShow').as('auth.signup.show')
```
Notice the slash at the end of the url. If I use `{{route('auth.signup.show')}}` I will get `/signup` without the slash.

Edit: in another example there is a trailing slash for some reason:

```
Route.group(() => {
    Route.get('/:page?/', 'PostsController.index')
      .as('index')
      .where('page', Route.matchers.number())
  })
    .prefix('/posts')
    .as('posts')
})
```
In this case `{{route('posts.index')}}` gives me `/posts/` with a slash, even if I remove the slash after `/:page?` parameter.

**Comment by aikrom** [comment]
By the way, it's not good idea handling `/post` and `/post/` as different routes. What's the difference in your routes in your case? What response do you want to return in each route?

**Reply by galaczi** [reply]
I don't want to handle them as different routes. None the less Google and the rest of the world does so it's bit of an SEO issue. Google was big on avoiding duplicate content and it's still a very important factor in rankings. Adding canonical is a must and probably addresses most of these concerns but even for users linking to the site it would be nice to have a clean url structure.
